Api

mubstimor opened the pull request:
A user can edit their orders
Included more tests.
Duplicate orders cannot be submitted.
Testing is done with more data.

mubstimor committed:
created new ui branch

mubstimor committed:
set up file for handling signup

mubstimor committed:
added interface for signup [Delivers #160197511]

mubstimor committed:
added interfaces for managing user classes [Delivers #160197517]

mubstimor committed:
added template interface for making orders [Delivers #160197517]

mubstimor committed:
Initial template for order history [Delivers #160197542]

mubstimor committed:
minor improvement for order interface [Delivers #160197517]

mubstimor committed:
Added order details template [Delivers #160197542]

mubstimor committed:
Initial template for viewing fast food items [Delivers #160197533]

mubstimor committed:
initial template for add/edititems [Delivers #160197524]

mubstimor committed:
initial set up for accepting, rejecting & completing orders. [Delivers #160197581 #160197591]

mubstimor committed:
Added details for marking orders as complete. [Delivers #160197591]

mubstimor committed:
updated css for sidebar

mubstimor committed:
Add css improvement for feature [Finishes #160197511]

mubstimor committed:
Css improvement for making orders & order history [Finishes #160197517 #160197542]

mubstimor committed:
Css improvement for making orders & order history [Finishes #160197581 #160197571 #160197542 #160197533 #160197524 #160197591]

mubstimor committed:
a little edit

mubstimor committed:
Create index.html for my project Github pages

mubstimor committed:
rollback on merge issue

mubstimor committed:
resolving merge conflict 2

mubstimor committed:
updated readme file

mubstimor committed:
fixed template responsiveness on making orders [Fixes #160197517]

mubstimor committed:
Css improvement for order history [Fixes #160197542]

mubstimor committed:
added Css responsiveness for admin dashboard

mubstimor committed:
improved template for accepting & declining orders [Fixes #160197581]

mubstimor committed:
extended food item to include price [Fixes #160197524]

mubstimor committed:
added delete effect to food item [Fixes #160197524]

mubstimor committed:
added responsiveness for customer order details

mubstimor committed:
setup initial base for api

mubstimor committed:
updated gitignore to remove .pyc files

mubstimor committed:
change to requirements on futures version

mubstimor committed:
added heroku link to readme

mubstimor committed:
changed response of the index method

mubstimor committed:
added travis.yml configurations

mubstimor committed:
removed test command from travis

mubstimor committed:
added script tag to travis

mubstimor committed:
adding test coverage to repo

mubstimor committed:
indentation fix for travis.yml

mubstimor committed:
bug fix for test_api.py

mubstimor committed:
bug fix for test_api.py index

mubstimor committed:
fixing pylint & coverage bugs

mubstimor committed:
setting correct version for futures in install

mubstimor committed:
add default config file for pylint

mubstimor committed:
populated pylint config

mubstimor committed:
change in command ordering

mubstimor committed:
fixing astroid version

mubstimor committed:
removed pylint from travis

mubstimor committed:
support branch deployment with travis ci to heroku

mubstimor committed:
fix auto deployment issue

mubstimor committed:
add git push to travis

mubstimor committed:
update gitignore to remove untracked files

mubstimor committed:
adding codecov badge

mubstimor committed:
fixing link for codecov badge

mubstimor committed:
added coveralls conf file

mubstimor committed:
added codeclimate keys to readme

mubstimor committed:
return request object for post order

mubstimor committed:
bug fix on response test for order

mubstimor committed:
post order via api successful [Delivers #160364205]

mubstimor committed:
test for post

mubstimor committed:
bug fix on test for post

mubstimor committed:
fixed bug on test for post

mubstimor committed:
improving code to achieve 100% score on pylint

mubstimor committed:
first implementation of CRUD [Delivers #160364208 #160364207 #160403099]

mubstimor committed:
completed tests for CRUD functions [Finishes #160397733]

mubstimor committed:
enforce pre-test conditions in tests

mubstimor committed:
seperating order functions from views

mubstimor committed:
removed unused codeclimate badges

mubstimor committed:
removed vscode files from git

mubstimor committed:
removed vscode files from git & gitignore

mubstimor committed:
changing coverage badge

mubstimor committed:
reverting to original badge

mubstimor committed:
added new package to supoort coverage

mubstimor committed:
removed codecov from travis

mubstimor committed:
delete coveralls conf file

mubstimor committed:
added coverage conf file

mubstimor committed:
readding coveralls.yml

mubstimor committed:
changing order for testing & coverage scripts

mubstimor committed:
changing coverage command

mubstimor committed:
changes to script commands

mubstimor committed:
set fixed coverage version

mubstimor committed:
removed coveralls token from git

mubstimor committed:
minor code improvements

mubstimor committed:
added user routes [Finishes #160458087 #160457670]

mubstimor committed:
added login function to user routes

mubstimor committed:
added tests for customer routes [Finishes #160459181]

mubstimor committed:
moving tests to their own package

mubstimor committed:
setting fooditem object aside

mubstimor committed:
fix slow update issue on coverage badge

mubstimor committed:
improving formatting of test code

mubstimor committed:
improved error handling on routes

mubstimor committed:
reducing parameters in update order route

mubstimor committed:
restricting acceptable values for status

mubstimor committed:
restructuring response for update order

mubstimor committed:
changed test for update order

mubstimor committed:
updated get user

mubstimor committed:
included more tests on create order

mubstimor committed:
testing with actual response data for create order

mubstimor committed:
added more tests to raise coverage score to atleast 98%

mubstimor committed:
point ui directly to its index page

mubstimor committed:
stopped addition of duplicate orders

mubstimor committed:
current status of api

mubstimor committed:
added endpoint for user updating their own orders

mubstimor committed:
incorporating test coverage badge try 1

mubstimor committed:
incorporating test coverage badge try 2

mubstimor committed:
incorporating test coverage badge try 3

mubstimor committed:
incorporating test coverage badge try 4

mubstimor committed:
added endpoint for retrieving user specific orders

mubstimor committed:
added support for order cancellation

mubstimor committed:
modified readme to show which endpoints have been implemented so far

mubstimor committed:
minor corrections

jeanjoe commented on README.md:
```diff
@@ -1,13 +1,53 @@
+PUT /orders/<orderId>
+
+* A user should be able to order for food
+POST /orders
```
I have tested your post endpoint and I get `"error": "Bad Request,     some parameters are either missing or invalid"` message and I do not know which data to provide, can you please add them to the README or return them in the response.

mubstimor replied:
sure thing, thanks. I'll update the readme

mubstimor replied:
For now, you can view the format of parameters in my test classes.

mubstimor committed:
added fooditem endpoints and modified readme to be more descriptive

jeanjoe commented:
Nice workflow, Just update your README.md

mubstimor commented:
Have updated it now, thank you so much for that feedback.

mubstimor committed:
merging ui changes to api

mubstimor committed:
added postgres setup requirements

mubstimor committed:
test setup for postgres db #1

mubstimor committed:
changing how environment variables are handled #2

mubstimor committed:
attempting to run with python 2.7

mubstimor committed:
undo create tables temporarily

mubstimor committed:
remove urlparse

mubstimor committed:
test db set up with travis

mubstimor committed:
added get menu endpoint [Delivers #160788487]

mubstimor committed:
added crud operations with persistence on food items  [Finishes #160788487]

mubstimor committed:
override get item test with a bigger id

mubstimor committed:
improve food item tests & database connection url

mubstimor committed:
added basic create-read operations for user with persistence [Delivers #160788461]

mubstimor committed:
added CRUD operations for orders with persistence [Delivers #160788465 #160788504 #160788481 #160788469 #160788476]

mubstimor committed:
tweaking test values

mubstimor committed:
tweaking test values 2

mubstimor committed:
improved tests for test order

mubstimor committed:
improved tests

mubstimor committed:
removed unused code & improved tests

mubstimor committed:
removed unnecessary commment code

mubstimor committed:
added basic starting point for jwt [Delivers #160788533]

mubstimor committed:
deleted unused manage.py file

mubstimor committed:
deleted unused files

mubstimor committed:
Merge pull request #6 from mubstimor/master

Transferring Api description file from apiary.io

mubstimor committed:
added documentation file - try 1 [Delivers #160848254]

mubstimor committed:
adding full url to apiary

mubstimor committed:
reversing change

mubstimor committed:
attempt to close db connections upon each request

mubstimor committed:
setting host for api

mubstimor committed:
added docs directory

mubstimor committed:
point travis to apiary branch [Delivers #160848254]

mubstimor committed:
show docs on heroku - try 2 [Delivers #160848254]

mubstimor committed:
added flasgger(for swagger documentation) to reqs

mubstimor committed:
removed unused depedency functools

mubstimor committed:
added sample documentation with swagger [Delivers #160848254]

mubstimor committed:
added jwt dependency

mubstimor committed:
docs(fastfoodfast): show update order

mubstimor committed:
docs(fastfoodfast): show update order

Add update order to the documentation page

[Delivers #160848254]

mubstimor committed:
Merge pull request #7 from mubstimor/docs

Adding documentation to API

mubstimor committed:
ft(fastfoodfast): add authentication with jwt

Restrict access to pages depending on role

[Delivers #160788533]

mubstimor committed:
ft(fastfoodfast): define routes to be authenticated

Add decorators to pages that should be accessed by logged in users

[Delivers #160788533]

mubstimor committed:
fix(fastfoodfast): add auth token to tests

Add authentication tokens to test headers

[Delivers #160955781]

mubstimor committed:
Merge pull request #8 from mubstimor/ft-jwt-auth-160788533

Ft jwt auth 160788533
